File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXSingleValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXSingleValue.h"


@implementation TDBXSingleValue
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:SingleType];
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	DBXInternalValue = 0;
}
-(bool) isNull {
	return ValueNull;
}
-(void) SetAsSingle:(float)value{
	DBXInternalValue = value;
}
-(float) GetAsSingle{
	return DBXInternalValue;
}

@end

File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXSingleValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap the Single type and allows it to be null
 *
 */


public class TDBXSingleValue  extends DBXValue {

	protected boolean ValueNull = false;
	private float DBXSingleValue;
	
	public TDBXSingleValue() {
		super();
		setDBXType(DBXDataTypes.SingleType);
	}
	
	public boolean isNull() {
		return ValueNull;
	}

	public void setNull() {
		ValueNull = true;
		DBXSingleValue = 0;
	}
	
	public void SetAsSingle(float Value) throws DBXException {
		if (!isNull())
			DBXSingleValue = Value;
	}

	public float GetAsSingle() throws DBXException {
		return DBXSingleValue;
	}

}